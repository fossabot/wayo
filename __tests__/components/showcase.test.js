import 'babel-polyfill';
import { shallowMount,createLocalVue } from '@vue/test-utils';
import { render } from '@vue/server-test-utils';
import Showcase from '../../src/components/showcase';
import ShowcaseBar from '../../src/components/showcase/_showcase-bar.vue';

const LocalVue = createLocalVue();

describe('Logic', () => {
  const Wrapper = shallowMount(Showcase, {
    localVue: LocalVue,
    slots: {
      default: ['<div></div>']
    }
  });

  test('navigator styles', () => {
    expect(Wrapper.vm.navStyles).toMatch(/height:\s*160px;/);

    Wrapper.setProps({
      navBg: '#ff0000'
    });
    expect(Wrapper.vm.navStyles).toMatch(/background\-color\:\s*#ff0000;/);
    expect(Wrapper.vm.navStyles).toMatch(/background\-image\:\s*none;/);
    Wrapper.setProps({
      navBg: 'http://map.sogou.com/logo.png'
    });
    expect(Wrapper.vm.navStyles).toMatch(/background\-image\:\s*url\(http\:\/\/map\.sogou\.com\/logo\.png\);/);
  });

  test('methods', () => {
    const Mock_show = jest.fn();
    const Mock_hide = jest.fn();

    Wrapper.setData({
      activeIndex: 1,
      panels: [{},{
        hide: Mock_hide
      },{
        show: Mock_show
      }]
    });
    Wrapper.vm.showItem(2);
    expect(Wrapper.vm.activeIndex).toBe(2);
    expect(Mock_hide).toBeCalled();
    expect(Mock_show).toBeCalled();

    Wrapper.setData({
      activeIndex: 1
    });
    Wrapper.vm.onBarClicked(2);
    expect(Wrapper.emitted()['active-change'][0]).toEqual([1]);
    expect(Mock_show.mock.calls.length).toBe(2);
    expect(Mock_hide.mock.calls.length).toBe(2);
  });
});

describe('Snapshot', () => {

  test('showcase', () => {
    const Wrapper = render(Showcase, {
      localVue: LocalVue,
      slots: {
        default: []
      }
    });
    expect(Wrapper).toMatchSnapshot();
  });

  test('showcase item', () => {
    const Wrapper = render(Showcase.Item, {
      localVue: LocalVue,
      propsData: {
        thumb: 'http://map.sogou.com/logo.png'
      }
    });
    expect(Wrapper).toMatchSnapshot();
  });

  test('showcase bar', () => {
    const Wrapper = render(ShowcaseBar, {
      localVue: LocalVue,
      propsData: {
        thumb: 'http://map.sogou.com/logo.png',
        width: 90,
        index: 0
      }
    });
    expect(Wrapper).toMatchSnapshot();
  });
});