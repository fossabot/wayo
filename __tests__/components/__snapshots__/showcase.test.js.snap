// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot showcase 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "wayo-showcase",
      "data-server-rendered": "true",
    },
    "children": Array [],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "data-server-rendered": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "data-server-rendered": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Snapshot showcase bar 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "wayo-showcase__navbar",
      "data-server-rendered": "true",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "wayo-showcase__thumb-box",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "wayo-showcase__thumb-img",
              "src": "http://map.sogou.com/logo.png",
              "style": "width:90px;visibility:hidden;",
            },
            "children": Array [],
            "name": "img",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "src": undefined,
              "style": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "src": undefined,
              "style": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "data-server-rendered": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "data-server-rendered": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`Snapshot showcase item 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "wayo-showcase__panel",
      "data-server-rendered": "true",
    },
    "children": Array [],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "data-server-rendered": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "data-server-rendered": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
